src/main_nep/parameters.cu

```diff
@@ -129,7 +130,8 @@ void Parameters::calculate_parameters()
       num_types * num_types * (dim_radial + n_max_angular + 1) * (basis_size_radial + 1);
   }
 
-  number_of_variables = number_of_variables_ann + number_of_variables_descriptor;
+  number_of_variables =
```

Comment:
make this within `if (version == 4)`

Reply:
That is, you need to make sure `number_of_variable` is still correct for NEP2 and NEP3.

Reply:
Right good point, forgot about that :+1: